sc/memiavl/multitree.go

```diff
@@ -414,6 +414,17 @@ func (t *MultiTree) Close() error {
 	return errors.Join(errs...)
 }
 
+func (t *MultiTree) ReplaceWith(other *MultiTree) error {
+	errs := make([]error, 0, len(t.trees))
+	for _, entry := range t.trees {
```

Comment:
I guess it's possible we happen to have store upgrades here, in that case current logic is not enough.

Reply:
Upgrades will already be applied to the MultiTree handled in ApplyUpgrades right? What else needs to be done here? Comparing to previous logic, I don't see we do anything special for upgrade during reload as well

Reply:
Are you saying there could be race condition between ReplaceWith and ApplyUpgrades as well? @yihuang

Reply:
I guess I was wrong, two trees should be identical here